### PR TITLE
Added HB-06 implementation

### PR DESCRIPTION
**Pull request by oroboro-cpu**

### CODE REVIEW

File: src/main/java/cinema/project/service/AuthenticationServiceImpl.java [hunk 1]
```diff
@@ -18,6 +20,7 @@ public User login(String email, String password) throws AuthenticationException
         if (user.isPresent()) {
             String hashPassword = HashUtil.hashPassword(password, user.get().getSalt());
             if (user.get().getPassword().equals(hashPassword)) {
+                shoppingCartService.registerNewShoppingCart(user.get());
```
Comment (IgorBochkor) [inline src/main/java/cinema/project/service/AuthenticationServiceImpl.java]: Checklist:
Use method registerNewShoppingCart() in AuthenticationServiceImpl class in register() method.

**Review by IgorBochkor — CHANGES_REQUESTED** [review]

**Review by IgorBochkor — APPROVED** [review]

File: src/main/java/cinema/project/dao/ShoppingCartDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<ShoppingCart> query =
+                    session.createQuery("SELECT s FROM ShoppingCart s"
+                            + " LEFT JOIN FETCH s.tickets t"
+                            + " LEFT JOIN FETCH t.movieSession m"
+                            + " LEFT JOIN FETCH m.movie"
+                            + " LEFT JOIN FETCH m.cinemaHall"
```
Comment (dmytro-hryhoruk) [inline src/main/java/cinema/project/dao/ShoppingCartDaoImpl.java:45]: do you need all of these joins?

File: src/main/java/cinema/project/service/AuthenticationServiceImpl.java [hunk 3]
```diff
@@ -29,6 +31,8 @@ public User register(String email, String password) {
         User user = new User();
         user.setEmail(email);
         user.setPassword(password);
-        return userService.add(user);
+        User user1 = userService.add(user);
+        shoppingCartService.registerNewShoppingCart(user1);
+        return user1;
```
Comment (dmytro-hryhoruk) [inline src/main/java/cinema/project/service/AuthenticationServiceImpl.java]: you can use the same user reference even without reassigning it. Check whether user object will have an id after userService.add()

**Review by dmytro-hryhoruk — CHANGES_REQUESTED** [review]

File: src/main/java/cinema/project/dao/ShoppingCartDaoImpl.java (hunk 2 above)
Comment (oroboro-cpu) [reply src/main/java/cinema/project/dao/ShoppingCartDaoImpl.java:45]: Yes, otherwise I get Lazy Initialization Exception

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**